tools/unsupervised_dataset/Crawler/CrawlerLog.py

```diff
@@ -0,0 +1,71 @@
+import datetime
+import csv
+import os
+
+class CrawlerLog:
+    """ The CrawlerLog keeps track of which websites were explored, how many queries were found,
+        and creates a CSV with all the queries. It also logs any errors encountered. The log is
+        saved into Logs subdirectory with name based on start time. Queries are saved into Queries
+        subdirectory.
+    """
+    
+    def __init__(self):
+        """ Initializes crawler log to keep track of crawler progress and instantiates
+            instance variables.
+        """
+        
+        self.startTime = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M")
+        
+        # Create directory for logs if it does not already exists
+        try:
+            os.mkdir("Logs")
+        except FileExistsError as e:
+            pass
```

Comment:
I am not expert in Python, is this recommended practice?
I think we usually check for existence before making directories, instead of ignoring exceptions?